src/test/java/com/google/api/codegen/testdata/csharp_gapic_resourcenames_library.baseline

```diff
@@ -0,0 +1,685 @@
+
+        /// <summary>
+        /// Parses a return resource name in string form into a <see cref="ReturnName"/>.
+        /// </summary>
+        /// <param name="returnName">The return resource name in string form. Must not be <c>null</c>.</param>
+        public static ReturnName Parse(string returnName)
+        {
+            GaxPreconditions.CheckNotNull(returnName, nameof(returnName));
+            ResourceName resourceName = s_template.ParseName(returnName);
+            return new ReturnName(resourceName[0], resourceName[1], resourceName[2]);
+        }
+
+        public static bool TryParse(string returnName, out ReturnName result)
+        {
+            GaxPreconditions.CheckNotNull(returnName, nameof(returnName));
+            ResourceName resourceName;
+            if (s_template.TryParseName(returnName, out resourceName))
+            {
+                result = new ReturnName(resourceName[0], resourceName[1], resourceName[2]);
+                return true;
+            }
+            else
+            {
+                result = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Constructs a return name from its component parts.
+        /// </summary>
+        /// <param name="shelfId">The shelf ID. Must not be <c>null</c>.</param>
+        /// <param name="bookId">The book ID. Must not be <c>null</c>.</param>
+        /// <param name="returnId">The return ID. Must not be <c>null</c>.</param>
+        public ReturnName(string shelfId, string bookId, string returnId)
+        {
+            ShelfId = GaxPreconditions.CheckNotNull(shelfId, nameof(shelfId));
+            BookId = GaxPreconditions.CheckNotNull(bookId, nameof(bookId));
+            ReturnId = GaxPreconditions.CheckNotNull(returnId, nameof(returnId));
+        }
+
+        /// <summary>
+        /// The shelf ID. Never <c>null</c>.
+        /// </summary>
+        public string ShelfId { get; }
+
+        /// <summary>
+        /// The book ID. Never <c>null</c>.
+        /// </summary>
+        public string BookId { get; }
+
+        /// <summary>
+        /// The return ID. Never <c>null</c>.
+        /// </summary>
+        public string ReturnId { get; }
+
+        /// <inheritdoc />
+        public override string ToString() => s_template.Expand(ShelfId, BookId, ReturnId);
+    }
+
+    public sealed partial class BookNameOneof : IResourceName
+    {
+        public enum Type
+        {
+            BOOK_NAME = 1,
+            ARCHIVED_BOOK_NAME = 2,
+            DELETED_BOOK_NAME_FIXED = 3,
+        }
+
+        public static BookNameOneof Parse(string name)
+        {
+            BookNameOneof result;
+            if (TryParse(name, out result))
+            {
+                return result;
+            }
+            throw new ArgumentException("Invalid name", "name");
+        }
+
+        public static bool TryParse(string name, out BookNameOneof result)
```

Comment:
I thought "fixed sets" would always accept a general resource name as well, so that the sets can be expanded.
Is that just still work to be done?